packages/cli/src/validators/schemas.ts

```diff
@@ -0,0 +1,341 @@
+import { JSONSchemaType } from 'ajv';
+import { CliConfig } from '../types/CliConfig';
+
+export const configFileSchema: JSONSchemaType<Partial<CliConfig>> = {
```

Comment:
there is a LOT of duplication in this file.

I would suggest splitting out a `tableSchema` and adding a function to make some fields nullable, similar to typescripts `Nullable<T>` type.

Reply:
This is what I originally tried to do, but for type breaks whenever you assign a passed variable to the JSONSchemaType